### PR TITLE
MM-38522 - POC: Digest of user's current tasks

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- First POC cut at a github-style todo list for assigned tasks
- This first version is not automated (that's this ticket: https://mattermost.atlassian.net/browse/MM-38668 )

After typing `/playbook todo` you get this dmed to you:
image

- will be doing e2e 

#### Progress: 
- POC (<--- we're here)
- Include list of ongoing runs
- Include list of overdue updates
- Send digest once a day when user logs on
- Checklist directly in the digest (custom post)
- Design polish (if needed)
- e2e tests (once we have a firmed up design)
- Telemetry (total number of in-notification link clicks, unique users who click)

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-38149
- Epic: https://mattermost.atlassian.net/browse/MM-38149

#### Checklist

- [ ] ~~Telemetry updated~~
- [ ] ~~Gated by experimental feature flag~~
- [x] Unit tests updated

### CODE REVIEW

File: server/sqlstore/playbook_run.go [hunk 1]
```diff
@@ -913,6 +913,61 @@ func (s *playbookRunStore) toPlaybookRun(rawPlaybookRun sqlPlaybookRun) (*app.Pl
...
+	}
+
+	query := s.store.builder.Select("i.ID AS PlaybookRunID", "i.Name AS PlaybookRunName",
+		"t.Name AS TeamName", "c.Name AS ChannelName", "c.DisplayName AS ChannelDisplayName",
+		"i.ChecklistsJSON AS ChecklistsJSON").
+		From("IR_Incident AS i").
+		Join("Teams AS t ON (i.TeamID = t.Id)").
+		Join("Channels AS c ON (i.ChannelID = c.Id)")
+	if s.store.db.DriverName() == model.DatabaseDriverMysql {
+		query = query.Where(sq.Like{"i.ChecklistsJSON": fmt.Sprintf("%%\"%s\"%%", userID)})
```
Comment (cpoile) [inline server/sqlstore/playbook_run.go:930]: MySQL has this more specific query:
```
query.Where(sq.Expr(fmt.Sprintf("JSON_CONTAINS(i.ChecklistsJSON -> '$[*].items[*].assignee_id', '\"%s\"')", userID)))
```
But Postgres 10 doesn't have anything similar. So doing this simple text search for both. As an aside, I'm almost certain that a simple text search is more performant for MySql anyway.

**Comment by lieut-data** [comment]
I'm going to hand off my primary review here, but happy to stay in the loop!

File: server/sqlstore/playbook_run.go [hunk 2]
```diff
@@ -913,6 +913,61 @@ func (s *playbookRunStore) toPlaybookRun(rawPlaybookRun sqlPlaybookRun) (*app.Pl
...
 }
 
+// GetAssignedTasks returns the list of tasks assigned to userID
+func (s *playbookRunStore) GetAssignedTasks(userID string) ([]app.AssignedRun, error) {
+	var raw []struct {
+		app.AssignedRun
+		ChecklistsJSON json.RawMessage
+	}
+
+	query := s.store.builder.Select("i.ID AS PlaybookRunID", "i.Name AS PlaybookRunName",
```
Comment (iomodo) [inline server/sqlstore/playbook_run.go:923]: Where we are getting all the runs where user has participated, which could be quite a lot. What do you think about getting only the active runs?

**Review by iomodo — APPROVED** [review]
Wonderful @cpoile, great work! Just a minor comment below.

File: server/sqlstore/playbook_run.go (hunk 2 above)
Comment (cpoile) [reply server/sqlstore/playbook_run.go:923]: Hah, nice catch, thanks @iomodo !

**Review by calebroseland — APPROVED** [review]

**Pull request merged**